sys/net/gnrc/sock/include/gnrc_sock_internal.h

```diff
@@ -55,6 +55,21 @@ extern "C" {
  */
 #define GNRC_SOCK_DYN_PORTRANGE_ERR (0)
 
+/**
+ * @brief   Structure to retrieve auxiliary data from @ref gnrc_sock_recv
+ *
+ * @details The members of this function depend on the modules used
+ * @internal
+ */
+typedef struct {
+#if IS_USED(MODULE_SOCK_AUX_LOCAL) || DOXYGEN
+    sock_ip_ep_t *local;    /**< local IP address PDU was received on */
+#endif
+#if !IS_USED(MODULE_SOCK_AUX_LOCAL)
+    uint8_t this_struct_is_not_empty;
+#endif
```

Comment:
Murdock wants this to be documented. Maybe just hide it from it?
```suggestion
#if !IS_USED(MODULE_SOCK_AUX_LOCAL) && !IS_ACTIVE(DOXYGEN)
    uint8_t this_struct_is_not_empty;
#endif
```

Reply:
I'm not sure if Doxygen is so smart that it can parse `IS_ACTIVE()` - it would need a C99 compliant preprocessor for this.

Maybe it is indeed the easiest to just document it.

Reply:
> I'm not sure if Doxygen is so smart that it can parse `IS_ACTIVE()` - it would need a C99 compliant preprocessor for this.

I think in that case it is just enough to check if it is not defined
```suggestion
#if !IS_USED(MODULE_SOCK_AUX_LOCAL) && !defined(DOXYGEN)
    uint8_t this_struct_is_not_empty;
#endif
```